### PR TITLE
raftstore: use new approximate keys function when result is 0

### PR DESCRIPTION
**Pull request by you06**
Signed-off-by: root <you1474600@gmail.com>

###  What have you changed?

Please explain in detail what the changes are in this PR and why they are needed:

- Summarize your change
When the new `get_region_approximate_keys_cf` function return 0, the `get_region_approximate_keys` function will downgrade to use the old `get_range_entries_and_versions` function. For this situation, 0 should be returned.

- Describe any limitations of the current code (optional)
Add a test but the the test also passed using old `get_range_entries_and_versions` function.

###  What is the type of the changes?

Pick one of the following and delete the others:

- Bugfix (a change which fixes an issue)

###  How is the PR tested?

Add a test function `raftstore::coprocessor::split_check::keys::tests::test_region_approximate_keys_sub_region`.

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

###  Refer to a related PR or issue link (optional)

#5306

### CODE REVIEW

**Comment by siddontang** [comment]
why the old get_range_entries_and_versions do not return 0? have we known it?

**Comment by you06** [comment]
@siddontang The old get_range_entries_and_versions using old points to get approximate range, for a region on small range, the old function may match a bigger range and produce a larger error than the new one.

File: src/raftstore/coprocessor/split_check/keys.rs [hunk 1]
```diff
@@ -386,4 +384,45 @@ mod tests {
...
+        let f = Box::new(MvccPropertiesCollectorFactory::default());
+        cf_opts.add_table_properties_collector_factory("tikv.mvcc-properties-collector", f);
+        let cfs_opts = LARGE_CFS
+            .iter()
+            .map(|cf| CFOptions::new(cf, cf_opts.clone()))
+            .collect();
+        let db = rocks::util::new_engine_opt(path_str, db_opts, cfs_opts).unwrap();
+
+        let cases = [("a", 1024), ("b", 2048), ("z", 4096)];
+        for &(key, vlen) in &cases {
```
Comment (hicqu) [inline src/raftstore/coprocessor/split_check/keys.rs]: What if write `b` and `c` and then flush? So they will be in same SST file.

Comment (you06) [reply src/raftstore/coprocessor/split_check/keys.rs]: flush only once now, this test will only pass `RangeProperties` approximate now.

**Comment by siddontang** [comment]
> The old get_range_entries_and_versions using old points to get approximate range, for a region on small range, the old function may match a bigger range and produce a larger error than the new one.

can we remove the old function directly?

**Comment by you06** [comment]
@siddontang It's not safe to remove old mvcc properties now, some users upgrade from old version may only have mvcc properties approximate points.

**Pull request merged**